src/main/presenter/workspacePresenter/ripgrepSearcher.ts

```
@@ -0,0 +1,163 @@
+      if (!proc.killed && proc.exitCode === null) {
+        proc.kill()
+      }
+      try {
+        const { code } = await exitPromise
+        // Exit code 0: matches found
+        // Exit code 1: no matches found (not an error)
+        // Exit code 2: error (e.g., "no files were searched" due to glob filter)
+        // For code 2, we've already logged stderr, and count is 0, so just return empty
+        // Only throw for unexpected errors (code > 2)
+        if (!terminatedEarly && code && code > 2) {
+          throw new Error(`Ripgrep exited with code ${code}: ${stderrOutput.substring(0, 200)}`)
```

Biome (2.1.2)

[error] Unsafe usage of 'throw'.

'throw' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)